samcli/lib/cookiecutter/interactive_flow.py

```diff
@@ -0,0 +1,92 @@
+"""A flow of questions to be asked to the user in an interactive way."""
+from typing import Any, Dict, Optional
+import click
+from .question import Question
+
+
+class InteractiveFlow:
```

Comment:
Question: **_Can we make the interactive part of sam init a library that can then be injected_**? does this have any external dependencies so is everything self contained within samcli/lib/cookiecutter

This way it be can like using `aws-lambda-builders` from `aws-sam-cli` through sam build. We can have clearly defined interfaces elsewhere and we only need to instantiate the appropriate interactive workflow.

Reply:
I like the idea of separating the flow of questions into its own class, instead of relying on multiple if, else statements. Nice!

Reply:
Yeah. this will eventually be wired up to a command right, `sam xyz` which will be an interactive experience, I want that part to be pluggable and driven by a library, and then we can possibly port over the same logic back to sam init.

Reply:
yeah, that is the intention of this effort.